src/vscode-dts/vscode.proposed.chatParticipantAdditions.d.ts

```diff
@@ -134,16 +134,25 @@ declare module 'vscode' {
 		constructor(uri: Uri, range: Range);
 	}
 
-	// Extended to add `SymbolInformation`. Would also be added to `constructor`.
 	export interface ChatResponseAnchorPart {
 		/**
 		 * The target of this anchor.
 		 *
 		 * If this is a {@linkcode Uri} or {@linkcode Location}, this is rendered as a normal link.
 		 *
 		 * If this is a {@linkcode SymbolInformation}, this is rendered as a symbol link.
+		 *
+		 * TODO mjbvz: Should this be a full `SymbolInformation`? Or just the parts we need?
+		 * TODO mjbvz: Should we allow a `SymbolInformation` without a location? For example, until `resolve` completes?
 		 */
 		value2: Uri | Location | SymbolInformation;
+
+		/**
+		 * Optional method which fills in the details of the anchor.
+		 *
+		 * THis is currently only implemented for symbol links.
+		 */
+		resolve?(token: CancellationToken): Thenable<void>;
```

Comment:
So the extension assigns the `resolve` function, and modifies the part when it's called?

Reply:
Yes that's how it's currently implemented. However we could make `resolve` return just the parts of the object that can be changed, such as a symbol kind and location